pkg/deploy/stack_deployer.go

```diff
@@ -54,35 +60,140 @@ type latticeServiceStackDeployer struct {
 	svcBuilder            gateway.LatticeServiceBuilder
 }
 
+var tgGcOnce sync.Once
+var tgGc *TgGc
+
 func NewLatticeServiceStackDeploy(
 	log gwlog.Logger,
 	cloud pkg_aws.Cloud,
 	k8sClient client.Client,
 ) *latticeServiceStackDeployer {
 	brTgBuilder := gateway.NewBackendRefTargetGroupBuilder(log, k8sClient)
 
+	tgMgr := lattice.NewTargetGroupManager(log, cloud)
+	tgSvcExpBuilder := gateway.NewSvcExportTargetGroupBuilder(log, k8sClient)
+	svcBuilder := gateway.NewLatticeServiceBuilder(log, k8sClient, brTgBuilder)
+
+	tgGcOnce.Do(func() {
+		// TODO: need to refactor TG synthesizer. Remove stack from constructor
+		// arguments and use it as Synth argument. That will help with Synth
+		// reuse for GC purposes
+		tgGcSynth := lattice.NewTargetGroupSynthesizer(log, cloud, k8sClient, tgMgr, tgSvcExpBuilder, svcBuilder, nil)
+		tgGcFn := NewTgGcFn(tgGcSynth)
+		tgGc = &TgGc{
+			lock:    sync.Mutex{},
+			log:     log.Named("gc"),
```

Comment:
log.Named("tgGc") ?

Reply:
usually log names are lower case, can do "tg_gc" or "tg.gc"

Reply:
Would we ever garbage collect services? I think this would be the only GC

Reply:
I think we should not use GC at all. It will be a problem in future - global lock will cause contention and need carefully lock/unlock otherwise we will deadlock route reconciliations. This solution is a little pain relief from doing full TG clean up on each reconciliation loop.

We should have a better mechanism to identify when we create brand new TG or when we replace old one and deletion is required. In that case we dont need locks, since problem will be properly partitioned by single-threaded owners - reconcilers.